x-pack/plugin/search-engines/src/main/java/org/elasticsearch/searchengines/action/CreateSearchEngineAction.java

```diff
@@ -0,0 +1,174 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.searchengines.action;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.ValidateActions;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.cluster.metadata.MetadataCreateIndexService;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.util.CollectionUtils;
+import org.elasticsearch.indices.InvalidEngineNameException;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class CreateSearchEngineAction extends ActionType<AcknowledgedResponse> {
+
+    public static final CreateSearchEngineAction INSTANCE = new CreateSearchEngineAction();
+    public static final String NAME = "indices:admin/search_engine/create";
+
+    private CreateSearchEngineAction() {
+        super(NAME, AcknowledgedResponse::readFrom);
+    }
+
+    public static class Request extends AcknowledgedRequest<Request> implements IndicesRequest {
+
+        private final String name;
+        private final long startTime;
+
+        private String[] indices;
+
+        private String relevanceSettingsId;
+
+        private static final ObjectParser<Request, RestRequest> PARSER;
+        static {
+            PARSER = new ObjectParser<>("create_search_engine");
+            PARSER.declareField(Request::setIndices, (p) -> {
+                List<String> indices = new ArrayList<>();
+                while ((p.nextToken()) != XContentParser.Token.END_ARRAY) {
+                    if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
+                        indices.add(p.text());
+                    }
+                }
+                return indices;
+            }, new ParseField("indices"), ObjectParser.ValueType.OBJECT_ARRAY);
+            PARSER.declareString(Request::setRelevanceSettingsId, new ParseField("relevance_settings_id"));
+        }
+
+        public Request(String name) {
+            this(name, System.currentTimeMillis(), new String[0], null);
+        }
+
+        public Request(String name, long startTime, String[] indices, String relevanceSettingsId) {
+            this.name = name;
+            this.startTime = startTime;
```

Comment:
Curious, why do we need the start time on the request? Is this a pattern in ES code?